sendgrid/resource_sendgrid_domain_authentication_validation.go

```diff
@@ -0,0 +1,104 @@
+/*
+Provide a resource to manage a domain authentication validation.
+Example Usage
+```hcl
+
+	resource "sendgrid_domain_authentication_validation" "foo" {
+		domain_authentication_id = sendgrid_domain_authentication.foo.id
+	}
+
+```
+*/
+package sendgrid
+
+import (
+	"context"
+	"fmt"
+
+	sendgrid "github.com/arslanbekov/terraform-provider-sendgrid/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// https://docs.sendgrid.com/api-reference/domain-authentication/validate-a-domain-authentication
+func resourceSendgridDomainAuthenticationValidation() *schema.Resource { //nolint:funlen
+	return &schema.Resource{
+		CreateContext: resourceSendgridDomainAuthenticationValidationCreate,
+		ReadContext:   resourceSendgridDomainAuthenticationValidationRead,
+		UpdateContext: resourceSendgridDomainAuthenticationValidationUpdate,
+		DeleteContext: resourceSendgridDomainAuthenticationValidationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"domain_authentication_id": {
+				Type:        schema.TypeString,
+				Description: "Id of the domain authentication to validate.",
+				Required:    true,
+				ForceNew:    true,
+			},
+
+			"valid": {
+				Type:        schema.TypeBool,
+				Description: "Indicates if this is a valid authenticated domain or not.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceSendgridDomainAuthenticationValidationCreate(
+	ctx context.Context,
+	d *schema.ResourceData,
+	m interface{},
+) diag.Diagnostics {
+	return validateDomain(ctx, d, m)
+}
+
+func validateDomain(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*sendgrid.Client)
+
+	if err := c.ValidateDomainAuthentication(ctx, d.Get("domain_authentication_id").(string)); err.Err != nil || err.StatusCode != 200 {
+		if err.Err != nil {
+			return diag.FromErr(err.Err)
+		}
+		return diag.Errorf("unable to validate domain DNS configuration")
+	}
+
```

Comment:
Here need to add the **ID** setting before calling **Read**
```suggestion

	d.SetId(domainAuthID)

```